NEWS.md

```diff
@@ -52,6 +52,7 @@ Standard library changes
 * Eigenvalues λ of general matrices are now sorted lexicographically by (Re λ, Im λ) ([#21598]).
 * `one` for structured matrices (`Diagonal`, `Bidiagonal`, `Tridiagonal`, `Symtridiagonal`) now preserves
   structure and type. ([#29777])
+* `diagm` can now construct a diagonal matrix from a vector (#[31125]).
```

Comment:
```suggestion
* `diagm(v)` is now a shorthand for `diagm(0 => v)` (#[31125]).
```
`diagm` has always been able to construct a diagonal matrix from a vector, it's just a question of how to spell it.